### DISCUSSION TITLE
Resource count over time by resource type/service etc?

### DISCUSSION THREAD

**Discussion by psilantropy** [microsoft/finops-toolkit#628] [Q&A] [answered]
Hi all.

Number of resources has a slight correlation to costs. It would be interesting to report on resource count over time.

Basically, I would like to show over time, although cost has reduced, we are deploying more than ever. Therefore, doing more with less.
This is semi-bs, but I think it's a nice metric to have in the toolkit. Has anybody done anything like this before?

I had a play with the CostSummary report (hub) and adjusted the services to count but I couldn't get the numbers right (new to PowerBI).
I don't really have any other places to report this data out of, and I think historical cost exports would be a great source.

Thanks

**Comment by flanakin** [comment, marked as answer]
I love this idea! There are a few challenges with resource counts in cost data vs. the Azure portal vs. Resource Graph. None of them will align for various reasons 😕 

> _Cost Management data only includes the usage and purchases from services and resources that are actively running. The cost data you see is based on past records. It includes resources, resource groups, and subscriptions that might be stopped, deleted, or canceled. So, it might not match with the current resources, resource groups, and subscriptions you see in tools like Azure Resource Manager or Azure Resource Graph. They only display currently deployed resources in your subscriptions. Not all resources emit usage and therefore might not be represented in the cost data. Similarly, Azure Resource Manager doesn't track some resources so they might not be represented in subscription resources._

https://learn.microsoft.com/en-us/azure/cost-management-billing/costs/understand-cost-mgt-data#:~:text=Cost%20Management%20data%20only,represented%20in%20subscription%20resources.

One more to add to this list is that some services emit different records in different cases, which Power BI treats as different resources when running unique counts. We don't lowercase it because casing is also important.

Want to create a feature request for this one? I'm also happy to take a PR for adding something, but Power BI PRs are difficult. Whatever is easier for you is great. I'm mainly curious what specific things you want to be able to see. I can add that to the backlog for 0.3.

**Reply by psilantropy** [reply]
Hi. Yes, will create one with a few suggestions. Thanks :).

That's what I was running in to after a play around. Ended up with too many records from different meters. But I think that metric (# of records emitting cost data) could still be useful, just unusual.

The only alternative I can find it just running a resource count every x and sending the result to csv or similar using logic apps.

**Reply by psilantropy** [reply]
**Resource Counts - New metric for reporting?** - https://github.com/microsoft/finops-toolkit/issues/632
